src/client/JoinPrivateLobbyModal.ts

```diff
@@ -261,6 +261,15 @@ export class JoinPrivateLobbyModal extends LitElement {
         }),
       );
 
+      return true;
+    } else if (archiveData.error === "Game not found") {
+      this.message = translateText("record.not_found");
+      return true;
+    } else if (archiveData.error === "Failed to read record") {
+      this.message = translateText("record.error");
+      return true;
+    } else if (archiveData.error === "Failed to parse record data") {
+      this.message = translateText("record.invalid_data");
```

Comment:
I recommend sending the translation key over the network.
```suggestion
    } else if (archiveData.error) {
      this.message = translateText(archiveData.error);
```